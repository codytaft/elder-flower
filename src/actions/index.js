export const setCurrentUser = user => ({
  type: 'SET_CURRENT_USER',
  user
});

export const createCarer = user => ({
  type: 'CREATE_CARER',
  user
});

export const createElder = user => ({
  type: 'CREATE_ELDER',
  user
});
