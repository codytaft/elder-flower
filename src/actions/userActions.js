export const setCurrentUser = user => ({
  type: 'SET_CURRENT_USER',
  user
});

export const createUser = user => ({
  type: 'CREATE_USER',
  user
});
