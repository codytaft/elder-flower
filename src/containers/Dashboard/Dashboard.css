.dashboard-wrapper {
  width: 100%; }

.navigation-bar {
  display: flex;
  flex-direction: row;
  justify-content: center;
  padding: 0;
  width: 100%;
  height: 5rem; }

.dashboard-calendar {
  margin: 3rem auto 3rem auto;
  width: 92%; }
