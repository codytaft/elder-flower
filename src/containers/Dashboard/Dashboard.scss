.navigation-bar {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 0;
  margin: 1rem 0 0 0;
  width: 100%;
  height: 5rem;
}
