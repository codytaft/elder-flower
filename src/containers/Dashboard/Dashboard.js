import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import Navigation from '../../components/Navigation/Navigation';
import { connect } from 'react-redux';
import { setCurrentUser } from '../../actions';

import './Dashboard.css';
export class Dashboard extends Component {
  render() {
    const { location } = this.props;
    return (
      <div className="dashboard-wrapper">
        <section className="navigation-bar">
          <Navigation />
        </section>
        {location.pathname === '/dashboard' ? (
          <iframe
            src="https://calendar.google.com/calendar/b/0/embed?showTitle=0&amp;showNav=0&amp;showDate=0&amp;showPrint=0&amp;showTabs=0&amp;showCalendars=0&amp;showTz=0&amp;mode=WEEK&amp;height=600&amp;wkst=1&amp;bgcolor=%23FFFFFF&amp;src=cody.taft%40gmail.com&amp;color=%2309d753&amp;ctz=America%2FDenver"
            width="800"
            height="600"
            frameBorder="0"
            scrolling="no"
            className="dashboard-calendar"
          />
        ) : (
          <div />
        )}
      </div>
    );
  }
}

Dashboard.propTypes = {
  setCurrentUser: PropTypes.func,
  currentUser: PropTypes.object,
  location: PropTypes.object
};

export const mapStateToProps = state => ({
  currentUser: state.currentUser
});

export const mapDispatchToProps = dispatch => ({
  setCurrentUser: user => dispatch(setCurrentUser(user))
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Dashboard)
);
