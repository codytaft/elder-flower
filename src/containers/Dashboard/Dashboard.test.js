import React from 'react';
import { shallow } from 'enzyme';
import { Dashboard, mapStateToProps, mapDispatchToProps } from './Dashboard';
import { setCurrentUser } from '../../actions';

describe('Dashboard', () => {
  it('should match the snapshot if pathname is dashboard', () => {
    let wrapper;
    const mockLocation = {
      pathname: '/dashboard',
      search: '',
      hash: ''
    };

    wrapper = shallow(<Dashboard location={mockLocation} />);
    expect(wrapper).toMatchSnapshot();
  });
});
it('should match the snapshot if pathname is not dashboard', () => {
  let wrapper;
  const mockLocation = {
    pathname: '/',
    search: '',
    hash: ''
  };

  wrapper = shallow(<Dashboard location={mockLocation} />);
  expect(wrapper).toMatchSnapshot();
});

describe('mapStateToProps', () => {
  it('should return an object with a currentUser object', () => {
    const mockCurrentUser = {
      id: 75,
      firstName: 'Cody',
      lastName: 'Taft',
      phoneNumber: '+19038511575',
      email: 'cody.taft@gmail.com',
      password: '123',
      contactName: 'Gaynell',
      contactPhone: '+17203304593',
      isElder: false
    };
    const mockState = {
      currentUser: { ...mockCurrentUser }
    };

    const expected = {
      currentUser: {
        id: 75,
        firstName: 'Cody',
        lastName: 'Taft',
        phoneNumber: '+19038511575',
        email: 'cody.taft@gmail.com',
        password: '123',
        contactName: 'Gaynell',
        contactPhone: '+17203304593',
        isElder: false
      }
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  it('should invoke dispatch when using a function from mapDispatchToProps', () => {
    const mockCurrentUser = {
      id: 75,
      firstName: 'Cody',
      lastName: 'Taft',
      phoneNumber: '+19038511575',
      email: 'cody.taft@gmail.com',
      password: '123',
      contactName: 'Gaynell',
      contactPhone: '+17203304593',
      isElder: false
    };
    const mockDispatch = jest.fn();
    const actionToDispatch = setCurrentUser(mockCurrentUser);

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setCurrentUser(mockCurrentUser);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
