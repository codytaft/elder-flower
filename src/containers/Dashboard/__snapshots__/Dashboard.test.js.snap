// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should match the snapshot if pathname is dashboard 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard
    location={
        Object {
            "hash": "",
            "pathname": "/dashboard",
            "search": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="navigation-bar"
>
          <Unknown />
</section>,
        <iframe
          className="dashboard-calendar"
          frameBorder="0"
          height="600"
          scrolling="no"
          src="https://calendar.google.com/calendar/b/0/embed?showTitle=0&showNav=0&showDate=0&showPrint=0&showTabs=0&showCalendars=0&showTz=0&mode=WEEK&height=600&wkst=1&bgcolor=%23FFFFFF&src=cody.taft%40gmail.com&color=%2309d753&ctz=America%2FDenver"
          width="800"
/>,
      ],
      "className": "dashboard-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Unknown />,
          "className": "navigation-bar",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "dashboard-calendar",
          "frameBorder": "0",
          "height": "600",
          "scrolling": "no",
          "src": "https://calendar.google.com/calendar/b/0/embed?showTitle=0&showNav=0&showDate=0&showPrint=0&showTabs=0&showCalendars=0&showTz=0&mode=WEEK&height=600&wkst=1&bgcolor=%23FFFFFF&src=cody.taft%40gmail.com&color=%2309d753&ctz=America%2FDenver",
          "width": "800",
        },
        "ref": null,
        "rendered": null,
        "type": "iframe",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="navigation-bar"
>
            <Unknown />
</section>,
          <iframe
            className="dashboard-calendar"
            frameBorder="0"
            height="600"
            scrolling="no"
            src="https://calendar.google.com/calendar/b/0/embed?showTitle=0&showNav=0&showDate=0&showPrint=0&showTabs=0&showCalendars=0&showTz=0&mode=WEEK&height=600&wkst=1&bgcolor=%23FFFFFF&src=cody.taft%40gmail.com&color=%2309d753&ctz=America%2FDenver"
            width="800"
/>,
        ],
        "className": "dashboard-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Unknown />,
            "className": "navigation-bar",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "dashboard-calendar",
            "frameBorder": "0",
            "height": "600",
            "scrolling": "no",
            "src": "https://calendar.google.com/calendar/b/0/embed?showTitle=0&showNav=0&showDate=0&showPrint=0&showTabs=0&showCalendars=0&showTz=0&mode=WEEK&height=600&wkst=1&bgcolor=%23FFFFFF&src=cody.taft%40gmail.com&color=%2309d753&ctz=America%2FDenver",
            "width": "800",
          },
          "ref": null,
          "rendered": null,
          "type": "iframe",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`should match the snapshot if pathname is not dashboard 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard
    location={
        Object {
            "hash": "",
            "pathname": "/",
            "search": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="navigation-bar"
>
          <Unknown />
</section>,
        <div />,
      ],
      "className": "dashboard-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Unknown />,
          "className": "navigation-bar",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="navigation-bar"
>
            <Unknown />
</section>,
          <div />,
        ],
        "className": "dashboard-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Unknown />,
            "className": "navigation-bar",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
